model-optimizer/extensions/middle/UnsqueezeTileReshapeBlockToInterpolate.py

```diff
@@ -113,14 +116,29 @@ def replace_pattern(self, graph: Graph, match: dict):
         shape_node.out_port(0).connect(strided_slice_node.in_port(0))
         slice_begin.out_port(0).connect(strided_slice_node.in_port(1))
         slice_end.out_port(0).connect(strided_slice_node.in_port(2))
-        strided_slice_node.out_port(0).connect(mul_node.in_port(0))
 
-        interp_node = Interpolate(graph, dict(name=unsqueeze_name + '/Interpolate_',
-                                              axes=int64_array([axis]),
-                                              mode='nearest')).create_node()
+        cast_shape_to_float = Cast(graph, {'dst_type': np.float32}).create_node()
```

Comment:
The same comment as for the previous file. Why the shape is converted to float?

Reply:
Fixed.